send.go

```diff
@@ -31,6 +31,17 @@ import (
 )
 
 func (c *Connection) RoundTrip(request *http.Request) (response *http.Response, err error) {
+	for _, callback := range c.preCallbacks {
+		request = callback.Pre(request)
+	}
+	response, err = c.roundTrip(request)
+	for _, callback := range c.postCallbacks {
+		response, err = callback.Post(response, err)
```

Comment:
Do you have an example to a list of post callbacks?

Theoretically, `err` could be changed inside the callback (depends on implementation), leading the following callback to act upon the previous callback error value.

Reply:
yes, an example would be a logging callback that logs the response status, and a metrics collecting callback to send data for analytics.